src/main/java/seedu/dengue/logic/commands/DeleteCommand.java

```diff
@@ -45,68 +42,45 @@ public class DeleteCommand extends Command {
     public static final String MESSAGE_DELETE_RANGE_SUCCESS = "%1$s cases from %2$s to %3$s deleted";
 
     private final Optional<List<Index>> targetIndexes;
-    private final Optional<Date> date;
-    private final Optional<Range<Date>> range;
-
-    // possibly separate into three different commands? index, date, range
+    private final Optional<DeleteDatePredicate> predicate;
```

Comment:
Actually, what is the rationale for making deletedatepredicate an optional? It can just be something that is always taken in, and if there are not prefixes that relates to date, the default is for the test?

Reply:
It helps with checking that index and predicate are not both present!